### PR TITLE
Add 1.7.1 and v1.6.5 to the Changelog

### PR DESCRIPTION
**Pull request by amartinezfayo**
Add 1.7.1 to the Changelog.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+
+### Added
+
+- x509pop node attestor emits a new selector with the leaf certificate serial number (#4216)
+- HTTPS server in the OIDC Discovery Provider can now be configured to use a certificate file (#4190)
+- Option to log source information in server and agent logs (#4246)
+
+### Changed
+
+- Improvement in entry lookup logic when fetching JWT-SVIDs (#4211)
```
Comment (azdagron) [inline CHANGELOG.md:13]: I dunno if this is worth mentioning since it is more or less just a code maintenance improvement?

**Review by azdagron — APPROVED** [review]
lgtm! just one optional comment

File: CHANGELOG.md (hunk 1 above)
Comment (evan2645) [reply CHANGELOG.md:13]: If it translates into a meaningful perf benefit, we could say something like "reduced CPU/memory usage when fetching JWT-SVIDs"

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+- Agent now has an exponential backoff strategy when syncing with the server (#4279)
+
+### Fixed
+
+- Bug distinguishing audience sets in the agent's JWT-SVID cache (#4309)
```
Comment (evan2645) [inline CHANGELOG.md:18]: We should probably be talking about user impact here, e.g.
```
- SPIRE Agent cache bug resulting in workloads receiving JWT-SVIDs with incomplete audience set
```

File: CHANGELOG.md (hunk 1 above)
Comment (amartinezfayo) [reply CHANGELOG.md:13]: It will probably not be a noticeable perf improvement in most cases. I'm now more inclined to remove this.

**Review by evan2645 — APPROVED** [review]

**Pull request merged**